config-model/src/main/java/com/yahoo/searchdefinition/ApplicationBuilder.java

```diff
@@ -201,28 +195,28 @@ public Schema add(Schema schema) {
         return schema;
     }
 
-    private Schema createSchema(String schemaString, String schemaPath) throws ParseException {
-        Schema schema = parseSchema(schemaString, schemaPath);
-        addRankProfileFiles(schema, schemaPath);
+    private Schema createSchema(String schemaString) throws ParseException {
+        Schema schema = parseSchema(schemaString);
+        addRankProfileFiles(schema);
         return schema;
     }
 
-    private Schema parseSchema(String schemaString, String schemaPath) throws ParseException {
+    private Schema parseSchema(String schemaString) throws ParseException {
         SimpleCharStream stream = new SimpleCharStream(schemaString);
         try {
             return new SDParser(stream, applicationPackage, fileRegistry, deployLogger, properties,
                                          rankProfileRegistry, documentsOnly)
-                    .schema(documentTypeManager, schemaPath);
+                    .schema(documentTypeManager);
         } catch (TokenMgrException e) {
             throw new ParseException("Unknown symbol: " + e.getMessage());
         } catch (ParseException pe) {
             throw new ParseException(stream.formatException(Exceptions.toMessageString(pe)));
         }
     }
 
-    private void addRankProfileFiles(Schema schema, String schemaPath) {
-        if (applicationPackage == null || schemaPath == null) return;
-        Path rankProfilePath = Path.fromString(schemaPath).append(schema.getName());
+    private void addRankProfileFiles(Schema schema) {
+        if (applicationPackage == null) return;
+        Path rankProfilePath = ApplicationPackage.SCHEMAS_DIR.append(schema.getName());
```

Comment:
Will not work with old schema dir (ApplicationPackage.SEARCH_DEFINITIONS_DIR), is that intended?

Reply:
Yes, but maybe it'll just cause confusion. I added it.